dotnet/src/SemanticKernel.Abstractions/Orchestration/IKernelContext.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel context reference
+/// </summary>
+public interface IKernelContext
+{
+    /// <summary>
+    /// App logger
+    /// </summary>
+    ILoggerFactory LoggerFactory { get; }
+
+    /// <summary>
+    /// Read only skills collection
+    /// </summary>
+    IReadOnlySkillCollection Skills { get; }
+
+    /// <summary>
+    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// </summary>
+    /// <param name="variables">Input to process</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <param name="pipeline">Target functions to run in sequence</param>
+    /// <returns>Result of the function composition</returns>
+    Task<SKContext> RunAsync(
+        ContextVariables variables,
+        CancellationToken cancellationToken = default,
+        params ISKFunction[] pipeline);
+
+    /// <summary>
+    /// Create a new instance of a context, linked to the kernel internal state.
+    /// </summary>
+    /// <param name="variables">Initializes the context with the provided variables</param>
+    /// <param name="skills">Provide specific scoped skills. Defaults to all existing in the kernel</param>
+    /// <returns>SK context</returns>
+    SKContext CreateNewContext(
+        ContextVariables? variables = null,
+        IReadOnlySkillCollection? skills = null);
+}
+
+/// <summary>
+/// Extension methods for <see cref="IKernelContext"/>
+/// </summary>
+public static class KernelContextExtensions
```

Comment:
Since a IKernel : IKernelContext, these extensions should also work for (and could replace) the similar methods inside of IKernel, right?

Reply:
Discussed offline. Resolving.